packages/postgres-database/src/simcore_postgres_database/utils_groups_extra_properties.py

```diff
@@ -0,0 +1,150 @@
+import datetime
+from dataclasses import dataclass, fields
+from typing import Any
+
+import sqlalchemy
+from aiopg.sa.connection import SAConnection
+from aiopg.sa.result import RowProxy
+from simcore_postgres_database.models.groups_extra_properties import (
+    groups_extra_properties,
+)
+
+from .models.groups import GroupType, groups, user_to_groups
+from .utils_models import FromRowMixin
+
+
+class GroupExtraPropertiesError(Exception):
+    ...
+
+
+class GroupExtraPropertiesNotFound(GroupExtraPropertiesError):
+    ...
+
+
+@dataclass(frozen=True, slots=True, kw_only=True)
+class GroupExtraProperties(FromRowMixin):
+    group_id: int
+    product_name: str
+    internet_access: bool
+    override_services_specifications: bool
+    created: datetime.datetime
+    modified: datetime.datetime
+
+
+async def _list_table_entries_ordered_by_group_type(
+    connection: SAConnection, user_id: int, product_name: str
+) -> list[RowProxy]:
+    subquery = (
+        sqlalchemy.select(
+            *[
```

Comment:
very nice subquery! :-)

Q: why you use expansion operations in the arguments as `sa.select(*[a,b])` instead of simply `sa.select(a,b)` ?

Reply:
it is because I am stupidly following chatty propositions. I will change it, although I am not sure that makes a huge difference.